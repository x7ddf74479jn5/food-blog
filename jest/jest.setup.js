// Optional: configure or set up a testing framework before each test.
// If you delete this file, remove `setupFilesAfterEnv` from `jest.config.js`

// Used for __tests__/testing-library.js
// Learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom/extend-expect";

window.matchMedia = jest.fn().mockImplementation((query) => {
  return {
    addListener: jest.fn(),
    matches: false,
    media: query,
    onchange: null,
    removeListener: jest.fn(),
  };
});
