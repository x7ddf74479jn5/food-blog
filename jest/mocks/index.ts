export { mockNextImage } from "./image";
export type { NextRouter } from "./router";
export { mockRouter, RouterContext, withMockedRouter } from "./router";
export { setupWindow } from "./window";
