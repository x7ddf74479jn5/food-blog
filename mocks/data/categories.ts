export const categories = {
  rice: { id: "1", slug: "rice", name: "ご飯物", color: "grey" },
  salad: { id: "2", slug: "salad", name: "サラダ", color: "grey" },
  tips: { id: "3", slug: "tips", name: "料理基礎", color: "grey" },
};
