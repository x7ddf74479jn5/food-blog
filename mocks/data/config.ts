import type { TConfig } from "@/types/models/config";

import { dateCommon } from "./utils";

export const mockConfig: TConfig = {
  ...dateCommon,
  apiHost: "https://food-blog.microcms.io/api/v1/",
  host: "https://food-blog-chi.vercel.app/",
  siteTitle: "Food Blog",
  siteDescription: "料理レシピを紹介するブログ",
  siteKeywords: ["料理"],
  siteImage: { url: "https://food-blog-chi.vercel.app/site_image.jpg", width: 100, height: 100 },
  organization: "Pandashark",
  officialSite: "https://next-portfolio-livid.vercel.app/",
  twitterId: "@pandashark6",
  email: "x7ddf74479jn5@gmail.com",
};
