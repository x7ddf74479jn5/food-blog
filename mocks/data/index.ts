export { mockArticles } from "./articles";
export { mockCategories } from "./categories";
export { mockConfig } from "./config";
export { mockPickup } from "./pickup";
export { mockTags } from "./tags";
export { dateCommon } from "./utils";
export { mockWriters } from "./writers";
