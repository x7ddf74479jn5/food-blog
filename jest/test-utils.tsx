import type Queries from "@testing-library/dom/types/queries";
import type { RenderResult } from "@testing-library/react";
import { render } from "@testing-library/react";
import type { RequestHandler } from "msw";
import { setupServer } from "msw/node";
import { SWRConfig } from "swr";

import { SearchProvider } from "@/contexts/search/SearchContext";
import { ViewportProvider } from "@/contexts/viewport/ViewportContext";

import { withMockLegacyRouter, withMockRouter } from "./mocks";

export const Providers: React.ComponentType<{ children?: React.ReactNode }> = ({ children }) => {
  return withMockRouter(
    withMockLegacyRouter(
      <SWRConfig value={{ dedupingInterval: 0, provider: () => new Map() }}>
        <SearchProvider>
          <ViewportProvider>{children}</ViewportProvider>
        </SearchProvider>
      </SWRConfig>
    )
  );
};

const customRender = (ui: React.ReactElement, options = {}): RenderResult<typeof Queries, HTMLElement> => {
  return render(ui, { wrapper: Providers, ...options });
};

// re-export everything
export * from "./mocks";
export * from "@testing-library/react";

// override render method
export { render as originalRender, customRender as render };

export const reTestCase = {
  anyImage: expect.stringMatching(/^(data:image\/gif)|\.(png|webp|jpeg|jpg|svg)$/),
  anyWord: expect.stringMatching(/\w+/),
};

type ErrorComponentProps = {
  message?: string;
};
export const ErrorComponent: React.FC<ErrorComponentProps> = ({ message }) => {
  throw new Error(message);
};

export const setupMockServer = (...handlers: RequestHandler[]) => {
  const server = setupServer(...handlers);
  beforeAll(() => server.listen());
  afterEach(() => server.resetHandlers());
  afterAll(() => server.close());
  return server;
};
