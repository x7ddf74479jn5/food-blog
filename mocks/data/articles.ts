import type { TArticle } from "@/types";

import { categories } from "./categories";
import { tags } from "./tags";
import { dateCommon } from "./utils";
import { writers } from "./writers";

const articleStock = {
  id: "1",
  title: "基本の一番だしの作り方",
  writer: writers.pandashark,
  description:
    "だし昆布と鰹節（削り節）から取った出汁は、味噌汁、そば、うどん、煮物、鍋料理など、多くの和食に使用できます。",
  image: { url: "/images/5026416_s.jpg", height: 427, width: 640 },
  excerpt:
    "だし昆布と鰹節（削り節）から取った出汁は、味噌汁、そば、うどん、煮物、鍋料理など、多くの和食に使用できます。",
  body: `**分量： 800ml**

だし昆布と鰹節（削り節）から取った出汁は、味噌汁、そば、うどん、煮物、鍋料理など、多くの和食に使用できます。

## 材料
- 水：1000ml
- だし昆布：10g
- かつお削り節：40g

## 作り方
- 固く絞った布巾で、昆布の表面の汚れを軽く拭き取ります。
- 鍋に水と昆布を入れます。
- 弱火で７分ほど熱し、鍋底から小さい泡が出てきたら昆布を取り出します。
- 昆布だしを一旦沸騰させてから、削り節を入れ、弱火で１分煮出します。
- ボウルの上にザルを置き、その上にキッチンペーパーを敷いて出汁を濾します。

---

**二番だしの取り方**
1. 鍋に 500ml の水を入れて沸騰させます。
2. 鍋に一番だしのだしがらを入れ、３分煮出します。
3. 一番だしと同じ方法で濾します。

---

## 保存方法
使わない分は保存容器に移して、冷蔵庫で保存してください。冷蔵保存で 2〜3 日は持ちます。

##  ポイント
だし昆布の表面の白い粉は「うま味」成分なので、取り除かずにそのままお使いください。`,
  category: categories.rice,
  tags: [tags.preparation, tags.misoSoup, tags.udon],
  imageOption: null,
};

type TArticleCollection = {
  [key: string]: TArticle;
};

export const articles: TArticleCollection = {
  stock: { ...dateCommon, ...articleStock },
};
