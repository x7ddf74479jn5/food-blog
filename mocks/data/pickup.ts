import { mockArticles } from "@mocks/data";
import { dateCommon } from "@mocks/data/utils";

export const mockPickup = {
  id: "1",
  name: "旬の食材を使ったおすすめレシピ",
  description: "旬の食材を使ったおすすめレシピ",
  startDate: new Date("2021/10/08 00:00"),
  endDate: new Date("2023/02/08 00:00"),
  articles: Object.values(mockArticles),
  ...dateCommon,
};
