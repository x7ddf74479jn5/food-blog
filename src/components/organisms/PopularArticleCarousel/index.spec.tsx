import { mockPopularArticles } from "@mocks/data";
import { render, screen } from "jest/test-utils";
import renderer from "react-test-renderer";

import { PopularArticleCarousel } from ".";

jest.mock("react-slick", () => {
  return {
    // eslint-disable-next-line @typescript-eslint/naming-convention
    __esModule: true,
    default: jest.fn((args) => {
      return <div>{args.children}</div>;
    }),
  };
});

describe("components/organisms/PopularArticleCarousel", () => {
  it("snapshot", () => {
    const tree = renderer.create(<PopularArticleCarousel articles={mockPopularArticles} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("OK: 初期表示が正しい", () => {
    const { container } = render(<PopularArticleCarousel articles={mockPopularArticles} />);
    expect(container).toHaveTextContent("POPULAR");
    expect(container).toHaveTextContent("人気記事ランキング");
    const articleEls = screen.getAllByRole("article");
    expect(articleEls.length).toBe(mockPopularArticles.length);
  });
});
