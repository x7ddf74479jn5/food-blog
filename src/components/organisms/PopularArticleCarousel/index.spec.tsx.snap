// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/organisms/PopularArticleCarousel snapshot 1`] = `
<section>
  <section>
    <div
      className="flex flex-row gap-2 items-center pl-1 mb-2"
    >
      <a
        href="/articles/popular"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <div
          className="flex flex-row gap-2 items-center text-xl"
        >
          <svg
            className="text-yellow-400"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M320 32H192a32 32 0 00-32 32v412a4 4 0 004 4h184a4 4 0 004-4V64a32 32 0 00-32-32zm144 160h-72a8 8 0 00-8 8v272a8 8 0 008 8h80a24 24 0 0024-24V224a32 32 0 00-32-32zM48 128a32 32 0 00-32 32v296a24 24 0 0024 24h80a8 8 0 008-8V136a8 8 0 00-8-8z"
            />
          </svg>
          <h2
            className="font-bold"
          >
            POPULAR
          </h2>
        </div>
      </a>
      <span
        className="flex-1 text-sm text-center break-words"
      >
        人気記事ランキング
      </span>
    </div>
    <div>
      <article
        className="px-2"
      >
        <div
          className="relative"
        >
          <a
            aria-label="基本の一番だしの作り方"
            href="/articles/p97vmuno3jdn"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <span
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "inline-block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "overflow": "hidden",
                  "padding": 0,
                  "position": "relative",
                  "width": "initial",
                }
              }
            >
              <span
                style={
                  Object {
                    "background": "none",
                    "border": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": "initial",
                    "margin": 0,
                    "maxWidth": "100%",
                    "opacity": 1,
                    "padding": 0,
                    "width": "initial",
                  }
                }
              >
                <img
                  alt=""
                  aria-hidden={true}
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27640%27%20height=%27360%27/%3e"
                  style={
                    Object {
                      "background": "none",
                      "border": 0,
                      "display": "block",
                      "height": "initial",
                      "margin": 0,
                      "maxWidth": "100%",
                      "opacity": 1,
                      "padding": 0,
                      "width": "initial",
                    }
                  }
                />
              </span>
              <img
                alt="Cover Image for 基本の一番だしの作り方"
                data-nimg="intrinsic"
                decoding="async"
                onError={[Function]}
                onLoad={[Function]}
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": "cover",
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
              <noscript>
                <img
                  alt="Cover Image for 基本の一番だしの作り方"
                  data-nimg="intrinsic"
                  decoding="async"
                  loading="lazy"
                  src="/_next/image?url=%2Fimages%2F5026416_s.jpg&w=1920&q=75"
                  srcSet="/_next/image?url=%2Fimages%2F5026416_s.jpg&w=640&q=75 1x, /_next/image?url=%2Fimages%2F5026416_s.jpg&w=1920&q=75 2x"
                  style={
                    Object {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": "cover",
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": 0,
                    }
                  }
                />
              </noscript>
            </span>
          </a>
          <div
            className="absolute top-0 left-0 p-2 text-sm font-semibold text-white bg-black/25"
          >
            1
          </div>
          <h3
            className="block absolute bottom-3 p-1 w-full text-sm font-semibold text-white truncate bg-black/25"
          >
            基本の一番だしの作り方
          </h3>
        </div>
      </article>
      <article
        className="px-2"
      >
        <div
          className="relative"
        >
          <a
            aria-label="新玉ねぎとトマトのシンプルサラダの作り方"
            href="/articles/mm8oec5icb"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <span
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "inline-block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "overflow": "hidden",
                  "padding": 0,
                  "position": "relative",
                  "width": "initial",
                }
              }
            >
              <span
                style={
                  Object {
                    "background": "none",
                    "border": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": "initial",
                    "margin": 0,
                    "maxWidth": "100%",
                    "opacity": 1,
                    "padding": 0,
                    "width": "initial",
                  }
                }
              >
                <img
                  alt=""
                  aria-hidden={true}
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27640%27%20height=%27360%27/%3e"
                  style={
                    Object {
                      "background": "none",
                      "border": 0,
                      "display": "block",
                      "height": "initial",
                      "margin": 0,
                      "maxWidth": "100%",
                      "opacity": 1,
                      "padding": 0,
                      "width": "initial",
                    }
                  }
                />
              </span>
              <img
                alt="Cover Image for 新玉ねぎとトマトのシンプルサラダの作り方"
                data-nimg="intrinsic"
                decoding="async"
                onError={[Function]}
                onLoad={[Function]}
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": "cover",
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
              <noscript>
                <img
                  alt="Cover Image for 新玉ねぎとトマトのシンプルサラダの作り方"
                  data-nimg="intrinsic"
                  decoding="async"
                  loading="lazy"
                  src="/_next/image?url=%2F2996666_s.jpg&w=1920&q=75"
                  srcSet="/_next/image?url=%2F2996666_s.jpg&w=640&q=75 1x, /_next/image?url=%2F2996666_s.jpg&w=1920&q=75 2x"
                  style={
                    Object {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": "cover",
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": 0,
                    }
                  }
                />
              </noscript>
            </span>
          </a>
          <div
            className="absolute top-0 left-0 p-2 text-sm font-semibold text-white bg-black/25"
          >
            2
          </div>
          <h3
            className="block absolute bottom-3 p-1 w-full text-sm font-semibold text-white truncate bg-black/25"
          >
            新玉ねぎとトマトのシンプルサラダの作り方
          </h3>
        </div>
      </article>
      <article
        className="px-2"
      >
        <div
          className="relative"
        >
          <a
            aria-label="栄養たっぷり小松菜と油揚げのおひたしの作り方"
            href="/articles/3y-w7i75kmd"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <span
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "inline-block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "overflow": "hidden",
                  "padding": 0,
                  "position": "relative",
                  "width": "initial",
                }
              }
            >
              <span
                style={
                  Object {
                    "background": "none",
                    "border": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": "initial",
                    "margin": 0,
                    "maxWidth": "100%",
                    "opacity": 1,
                    "padding": 0,
                    "width": "initial",
                  }
                }
              >
                <img
                  alt=""
                  aria-hidden={true}
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27640%27%20height=%27360%27/%3e"
                  style={
                    Object {
                      "background": "none",
                      "border": 0,
                      "display": "block",
                      "height": "initial",
                      "margin": 0,
                      "maxWidth": "100%",
                      "opacity": 1,
                      "padding": 0,
                      "width": "initial",
                    }
                  }
                />
              </span>
              <img
                alt="Cover Image for 栄養たっぷり小松菜と油揚げのおひたしの作り方"
                data-nimg="intrinsic"
                decoding="async"
                onError={[Function]}
                onLoad={[Function]}
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": "cover",
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
              <noscript>
                <img
                  alt="Cover Image for 栄養たっぷり小松菜と油揚げのおひたしの作り方"
                  data-nimg="intrinsic"
                  decoding="async"
                  loading="lazy"
                  src="/_next/image?url=%2F4966941_s.jpg&w=1920&q=75"
                  srcSet="/_next/image?url=%2F4966941_s.jpg&w=640&q=75 1x, /_next/image?url=%2F4966941_s.jpg&w=1920&q=75 2x"
                  style={
                    Object {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": "cover",
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": 0,
                    }
                  }
                />
              </noscript>
            </span>
          </a>
          <div
            className="absolute top-0 left-0 p-2 text-sm font-semibold text-white bg-black/25"
          >
            3
          </div>
          <h3
            className="block absolute bottom-3 p-1 w-full text-sm font-semibold text-white truncate bg-black/25"
          >
            栄養たっぷり小松菜と油揚げのおひたしの作り方
          </h3>
        </div>
      </article>
      <article
        className="px-2"
      >
        <div
          className="relative"
        >
          <a
            aria-label="熊本県産玉ねぎを使った絶品タコライスの作り方"
            href="/articles/q_y-lyvhew"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <span
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "inline-block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "overflow": "hidden",
                  "padding": 0,
                  "position": "relative",
                  "width": "initial",
                }
              }
            >
              <span
                style={
                  Object {
                    "background": "none",
                    "border": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": "initial",
                    "margin": 0,
                    "maxWidth": "100%",
                    "opacity": 1,
                    "padding": 0,
                    "width": "initial",
                  }
                }
              >
                <img
                  alt=""
                  aria-hidden={true}
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27640%27%20height=%27360%27/%3e"
                  style={
                    Object {
                      "background": "none",
                      "border": 0,
                      "display": "block",
                      "height": "initial",
                      "margin": 0,
                      "maxWidth": "100%",
                      "opacity": 1,
                      "padding": 0,
                      "width": "initial",
                    }
                  }
                />
              </span>
              <img
                alt="Cover Image for 熊本県産玉ねぎを使った絶品タコライスの作り方"
                data-nimg="intrinsic"
                decoding="async"
                onError={[Function]}
                onLoad={[Function]}
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": "cover",
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
              <noscript>
                <img
                  alt="Cover Image for 熊本県産玉ねぎを使った絶品タコライスの作り方"
                  data-nimg="intrinsic"
                  decoding="async"
                  loading="lazy"
                  src="/_next/image?url=%2F3129120_s.jpg&w=1920&q=75"
                  srcSet="/_next/image?url=%2F3129120_s.jpg&w=640&q=75 1x, /_next/image?url=%2F3129120_s.jpg&w=1920&q=75 2x"
                  style={
                    Object {
                      "border": "none",
                      "bottom": 0,
                      "boxSizing": "border-box",
                      "display": "block",
                      "height": 0,
                      "left": 0,
                      "margin": "auto",
                      "maxHeight": "100%",
                      "maxWidth": "100%",
                      "minHeight": "100%",
                      "minWidth": "100%",
                      "objectFit": "cover",
                      "objectPosition": undefined,
                      "padding": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": 0,
                    }
                  }
                />
              </noscript>
            </span>
          </a>
          <div
            className="absolute top-0 left-0 p-2 text-sm font-semibold text-white bg-black/25"
          >
            4
          </div>
          <h3
            className="block absolute bottom-3 p-1 w-full text-sm font-semibold text-white truncate bg-black/25"
          >
            熊本県産玉ねぎを使った絶品タコライスの作り方
          </h3>
        </div>
      </article>
    </div>
  </section>
</section>
`;
